### PR TITLE
feat(fulfillment): Events management

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
This pr includes some cleanup and refactoring around the abstract event emitter method for the module service to not rely on a named class property but instead on resolution naming.

Includes fulfillment set events on creation. 

The idea is that if that pr allows us to align and agreed on the approach as well as including the cleanup for other pr to use, If this gets merged I ll continue with another pr to do the rest of the event management

partially fix CORE-1735

### CODE REVIEW

File: packages/fulfillment/src/services/fulfillment-module-service.ts [hunk 1]
```diff
@@ -1557,4 +1566,66 @@ export default class FulfillmentModuleService<
...
+    for (const fulfillmentSet of createdFulfillmentSets) {
+      sharedContext.messageAggregator!.saveRawMessageData(
+        buildMessage({
+          eventName: FulfillmentUtils.FulfillmentEvents.created,
+          id: fulfillmentSet.id,
+          object: "fulfillment_set",
+        })
+      )
+
+      if (fulfillmentSet.service_zones?.length) {
```
Comment (sradevski) [inline packages/fulfillment/src/services/fulfillment-module-service.ts]: We can have an early return here, and below as well

**Review by sradevski — APPROVED** [review]
LGTM

File: packages/utils/src/event-bus/utils.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import { camelToSnakeCase, kebabCase, lowerCaseFirst } from "../common"
```
Comment (olivermrbl) [inline packages/utils/src/event-bus/utils.ts:1]: praise: nice!

File: packages/fulfillment/src/services/fulfillment-module-service.ts (hunk 1 above)
Comment (adrien2p) [reply packages/fulfillment/src/services/fulfillment-module-service.ts]: simplified a bit more

Comment (adrien2p) [reply packages/fulfillment/src/services/fulfillment-module-service.ts]: wdyt?

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**